rust/main/hyperlane-base/src/settings/base.rs

```diff
@@ -215,13 +215,12 @@ impl Settings {
     /// watermark trait bounds
     pub async fn contract_syncs<T, S>(
         &self,
-        domains: impl Iterator<Item = &HyperlaneDomain>,
         metrics: &CoreMetrics,
         sync_metrics: &ContractSyncMetrics,
         stores: HashMap<HyperlaneDomain, Arc<S>>,
         advanced_log_meta: bool,
         broadcast_sender_enabled: bool,
-    ) -> Result<HashMap<HyperlaneDomain, Arc<dyn ContractSyncer<T>>>>
+    ) -> HashMap<HyperlaneDomain, eyre::Result<Arc<dyn ContractSyncer<T>>>>
```

Comment:
so I understand this PR correctly: this is the most significant thing changed, where instead of propagating an Err we will handle it more gracefully?

Reply:
yeah, we would handle it in the top level in the relayer code instead of here